docs/source/01_key_concepts.rst

```diff
@@ -0,0 +1,79 @@
+============
+Key Concepts
+============
+
+-----------
+Time series
+-----------
+
+A time series is a sequence of data indexed by timestamps, which can be univariate or multivariate as long as the index is shared across variables. In TStore, a time series is represented by a `TS` object.
+
+-------
+TSArray
+-------
+
+A `TSArray` is an array of `TS` objects. It is actually [a pandas `ExtensionArray` which is used to support custom data types](https://pandas.pydata.org/docs/reference/api/pandas.api.extensions.ExtensionArray.html). For instance, given a list of stations, a `TSArray` can be used to map each station to a `TS` object with the corresponding time series of observed data.
+
+----
+TSDF
+----
+
+A `TSDF` is a table-like object where each column is a `TSArray`. It is a convenient way to map multiple time series (e.g., multiple variables) to a common index (e.g., a given station). The following diagram illustrates how a `TSDF` object is composed of `TSArray` objects, which are themselves composed of `TS` objects:
+
+.. image:: static/tsdf.svg
+  :width: 400
+  :alt: TSDF
+
+**When is this useful?** Consider the case of meteorological observations, where different stations may feature different sensors to record different variables, e.g., temperature, precipitation, wind speed, etc. Stations may have different observation frequencies, periods of inoperation or missing data and the like. Therefore it may not be appropriate to store all the observations within the same multivariate time series, as it could result in many NaN values - some because of differences in observation frequency, some because of missing data. Instead, the observations can be structured as a `TSArray` with a dedicated `TS` object for each station.
+
+Additionally, it may be appropriate to store the observations of different variables in separate `TSArray` objects, which can be combined in a `TSDF` object. This is particularly useful when a station is equipped with multiple sensors to record different variables, which may have different observation frequencies. In fact, variables such as precipitation may have long periods without records, which would result in many NaN values in a multivariate time series.
+
+------
+TSLong
+------
+
+In a **long-form data table**, each variable corresponds to a column and each observation corresponds to a row.
```

Comment:
I would expand this: 
- duplicated timestamps (when simultaneous measurements at multiple stations) 
- lot of NaN if irregular data 
- how to deal with temporal resolutions  
- ...